game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -2313,39 +2230,31 @@ public static void triggerActivateTriggerOther(final Map<ICondition, Boolean> te
    * Triggers all victory notifications associated with {@code satisfiedTriggers}.
    */
   public static void triggerVictory(final Set<TriggerAttachment> satisfiedTriggers, final IDelegateBridge bridge,
-      final String beforeOrAfter, final String stepName, final boolean useUses, final boolean testUses,
-      final boolean testChance, final boolean testWhen) {
+      final FireTriggerParams fireTriggerParams) {
 
     // NOTE: The check is needed to ensure that UI-related code (namely 'NotificationMessages.getInstance()') is
     // not executed when unit testing non-UI related code such as 'MustFightBattleTest'.
     if (satisfiedTriggers.stream().anyMatch(victoryMatch())) {
       triggerVictory(
-          satisfiedTriggers, bridge, beforeOrAfter, stepName, useUses, testUses, testChance, testWhen,
+          satisfiedTriggers, bridge, fireTriggerParams,
           NotificationMessages.getInstance());
     }
   }
 
   @VisibleForTesting
   static void triggerVictory(final Set<TriggerAttachment> satisfiedTriggers, final IDelegateBridge bridge,
-      final String beforeOrAfter, final String stepName, final boolean useUses, final boolean testUses,
-      final boolean testChance, final boolean testWhen,
-      final NotificationMessages notificationMessages) {
+      final FireTriggerParams fireTriggerParams, final NotificationMessages notificationMessages) {
     final GameData data = bridge.getData();
-    Collection<TriggerAttachment> trigs = CollectionUtils.getMatches(satisfiedTriggers, victoryMatch());
-    if (testWhen) {
-      trigs = CollectionUtils.getMatches(trigs, whenOrDefaultMatch(beforeOrAfter, stepName));
-    }
-    if (testUses) {
-      trigs = CollectionUtils.getMatches(trigs, availableUses);
-    }
+    final Collection<TriggerAttachment> trigs = filterSatisfiedTriggers(
+        satisfiedTriggers, victoryMatch(), fireTriggerParams);
     for (final TriggerAttachment t : trigs) {
-      if (testChance && !t.testChance(bridge)) {
+      if (fireTriggerParams.testChance && !t.testChance(bridge)) {
         continue;
       }
-      if (useUses) {
+      if (fireTriggerParams.useUses) {
         t.use(bridge);
       }
-      if (t.getVictory() == null || t.getPlayers() == null) {
+      if (t.getPlayers() == null) {
```

Comment:
I think you need to clarify this change, is `getVictory()` guaranteed to be non-null?

Reply:
Very good point. I removed it because that check is contained inside `victoryMatch`, which is used to filter those `TriggerAttachment`s before the `for`-loop. I believe that the code inside the `for`-loop does not at any point modify `victory`, though I must admit that I am not fully certain about it.